### ISSUE TITLE
Plesk server: cURL error: 6 - Could not resolve host:

### ISSUE THREAD

**Issue by kearfy** [boxbilling/boxbilling#841]
Hi there,

I am trying to set up boxbilling with Plesk, but it does not work. Returns the error stated in the title.
I did a bit of digging around and it seems as if all the properties in $this->_config are NULL. Thus a non existant host (:8443/url/bla/bla).

When I change the server manager to something like cpanel it works flawlessly. Did a total of 2 installs and I have done a bit of digging around for myself but I cannot find a solution. 

![image](https://user-images.githubusercontent.com/17960250/107412180-aa40ff80-6b0f-11eb-87ec-83b0129ebeb6.png)

**Comment by BenNottelling** [comment]
@timothygwebb You said you used Plesk with BoxBilling, right? Do you have any insight into this problem?

**Comment by timothygwebb** [comment]
Not able to resolve DNS make sure nameservers in resolv.conf are accessible
and restart service for networking this is not a BoxBilling issue curl as
well has a known issue with Plesk. Simple google search for error will give
instructions if need be for "plesk curl error 6 could not resolve host"

Thanks

Timothy Webb Sr.

**Comment by BenNottelling** [comment]
Could you please try again with the latest version of this server manager?
Thanks. 
https://github.com/boxbilling/boxbilling/blob/master/src/bb-library/Server/Manager/Plesk.php

**Comment by wassi007** [comment]
Hello, 
I also have the curl error6 problem with Plesk. The current version of plesk.php did not help either.
I noticed that in the plesk.php script in the function "getLoginUrl" the variable $this->_config['host'] is passed. But this is empty. The URL of the Plesk host is in $this->_host['host'].
After replacing "_config" with "_host" in lines 25, 28, 578 & 579, I got the output "Success connected" and was also able to create products and enter orders where subscriptions were created in Plesk.
Can I dare to continue working like this or can problems occur by changing "_config" to "_host"?
Many greetings
Jürgen
